### ISSUE TITLE
False negative: Tuple index out of range

### ISSUE THREAD

**Issue by goodspark** [python/mypy#10317]
**Bug Report**

Getting a false negative on tuple indices and lengths.

**To Reproduce**

```py3
from typing import Tuple, Union

def zxc(x: Union[Tuple[str], Tuple[str, int]]) -> None:
    if len(x) == 1:
        print(x[0])
    elif len(x) == 2:
        print(x[0], x[1])  # line 7
```

```sh
mypy mypytest.py
```

**Expected Behavior**

Expected mypy to pass. As far as I can tell, the types are correct.

**Actual Behavior**

Mypy fails with:
```
mypytest.py:7: error: Tuple index out of range
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.812
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: 3.8.5
- Operating system and version: Ubuntu 20.04

**Comment by hauntsaninja** [comment]
Duplicate of #9171 , #1178 and others (also this is a false positive :-) )
